### PR TITLE
[*] Logger - Improve formatting, add stack if present, do not display error messages twice

### PR DESCRIPTION
**Pull request by rap2hpoutre**
### Description

The request has a bit changed. I tried to fix the actual behavior of `logger`:
- Add space where needed (no more `Unexpected error:Using this message`)
- Display stack trace if present.
- Change code where error message was displayed twice (remove `error.message` in first string).

Now, this code:

```javascript
logger.error('Error');

try {
  throw new Error('Invalid column name `id`.');
} catch (e) {
  logger.error('Unexpected error: ', e);
}

logger.error('Unexpected error: my message');

try {
  throw new Error('Invalid column name `id`.');
} catch (e) {
  logger.error(`Unexpected error: ${e.message}`, e);
}

logger.info('hello', {});

try {
  throw new Error('Invalid column name `id`.');
} catch (e) {
  logger.warn('Foobar. ', e);
}
```

Displays:

```
[forest] 🌳🌳🌳  Error
[forest] 🌳🌳🌳  Unexpected error: Invalid column name `id`.
Error: Invalid column name `id`.
    at Object.<anonymous> (/Users/raphaelh/web/forest-express/test-logger.js:6:9)
    at Module._compile (internal/modules/cjs/loader.js:956:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:973:10)
    at Module.load (internal/modules/cjs/loader.js:812:32)
    at Function.Module._load (internal/modules/cjs/loader.js:724:14)
    at Function.Module.runMain (internal/modules/cjs/loader.js:1025:10)
    at internal/main/run_main_module.js:17:11
[forest] 🌳🌳🌳  Unexpected error: my message
[forest] 🌳🌳🌳  Unexpected error: Invalid column name `id`.Invalid column name `id`.
Error: Invalid column name `id`.
    at Object.<anonymous> (/Users/raphaelh/web/forest-express/test-logger.js:14:9)
    at Module._compile (internal/modules/cjs/loader.js:956:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:973:10)
    at Module.load (internal/modules/cjs/loader.js:812:32)
    at Function.Module._load (internal/modules/cjs/loader.js:724:14)
    at Function.Module.runMain (internal/modules/cjs/loader.js:1025:10)
    at internal/main/run_main_module.js:17:11
[forest] 🌳🌳🌳  hello
[forest] 🌳🌳🌳  Foobar. Invalid column name `id`.
Error: Invalid column name `id`.
    at Object.<anonymous> (/Users/raphaelh/web/forest-express/test-logger.js:22:9)
    at Module._compile (internal/modules/cjs/loader.js:956:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:973:10)
    at Module.load (internal/modules/cjs/loader.js:812:32)
    at Function.Module._load (internal/modules/cjs/loader.js:724:14)
    at Function.Module.runMain (internal/modules/cjs/loader.js:1025:10)
    at internal/main/run_main_module.js:17:11
```

## Pull Request checklist:

- [x] Write an explicit title for the Pull Request
- [x] Write changes made in the CHANGELOG.md
- [x] Test manually the implemented changes
- [x] Review my own code (indentation, syntax, style, simplicity, readability)
- [x] Wonder if you can improve the existing code

### CODE REVIEW

**Comment by rap2hpoutre** [comment]
[Error messages are displayed twice in console (using a fresh Lumber project)](https://trello.com/c/t0cgkYrT/3920-error-messages-are-displayed-twice-in-console-using-a-fresh-lumber-project)

File: src/index.js [hunk 1]
```diff
@@ -58,15 +58,15 @@ function requireAllModels(modelsDir) {
         recursive: true,
       });
     } catch (error) {
-      logger.error(`Cannot read a file for the following reason: ${error.message}`, error);
```
Comment (arnaudbesnier) [inline src/index.js]: I think the `error` printed after the error message is supposed to give some context (stack trace,...).
So I am wondering if we really want to do that?

File: src/index.js [hunk 2]
```diff
@@ -58,15 +58,15 @@ function requireAllModels(modelsDir) {
...
+      logger.error(`Cannot read a file for the following reason: ${error.message}`);
     }
   }
 
   // NOTICE: User didn't provide a modelsDir but may already have required them manually so they
   //         might be available.
   return P.resolve(getModels())
     .catch((error) => {
-      logger.error(`Cannot read a file for the following reason: ${error.message}`, error);
+      logger.error(`Cannot read a file for the following reason: ${error.message}`);
```
Comment (arnaudbesnier) [inline src/index.js]: Same comment here.

File: src/services/error-handler.js [hunk 3]
```diff
@@ -10,7 +10,7 @@ exports.catchIfAny = (error, request, response, next) => {
 
     if (!error.status) {
       // NOTICE: Unexpected errors should log an error in the console.
-      logger.error(`Unexpected error: ${message}`, error);
+      logger.error(`Unexpected error: ${message}`);
```
Comment (arnaudbesnier) [inline src/services/error-handler.js]: Same comment here too. Especially for unexpected errors we need context for debug purpose.

Can you explain how it fixes the issue described in the ticket, I don't really understand at first sight.
Capture_d’écran_2019-11-29_à_08 30 14

Thanks!

**Comment by rap2hpoutre** [comment]
Updated description. ☝️

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,6 +1,8 @@
 # Change Log
 
 ## [Unreleased]
+### Changed
```
Comment (arnaudbesnier) [inline CHANGELOG.md]: I would not be offended if you said it is a fix (because clearly it is) ;)

**Pull request merged**